src/game/MatrixGeneration.ts

```diff
@@ -0,0 +1,249 @@
+import GameConfiguration from "./GameConfiguration";
+
+import seedrandom from "seedrandom";
+
+//TODO should this defined here?
+//TODO how to choose these numbers?
+//TODO is this correct way to provide constants? (what are the naming conventions?)
+const MinMatrixSize = 3;
+const MaxMatrixSize = 8;
+
+const MinNumberSequences = 2;
+const MaxNumberSequences = 7;
+const MinSequenceLength = 2;
+
+const MinBufferSize = 4;
+const MaxBufferSize = 8;
+
+const StepsUntilMatrixSizeIncreased = 10;
+const StepsUntilSequenceNumberIncreased = 3;
+
+const DefaultTimeout = 60_000;
+
+//TODO see how many different values are there
+//TODO see how the possible values are defined
+const numberDifferentMatrixValues = 5;
+const matrixValues = ["A0", "E9", "4C", "8B", "6F"];
+const EmptyMatrixValue = "  ";
+
+/**
+ * Generates a randomly filled matrix and according sequences given the level and
+ * a seed.
+ * 
+ * The matrix is always saved in the first array of the function's result.
+ * I.e., result[0] is the matrix, while all other arrays contain sequences.
+ * 
+ * @param level The current game level (corresponds directly to the difficulty)
+ * @param seed  The seed used to compute the random values of the marix and sequences
+ */
+export function generateGameConfig(level: number, seed: string): GameConfiguration {
+    //the levels probably start at 1, however starting with 0 makes things smoother here
```

Comment:
Check boundaries (throw on level < 0 for example).

Reply:
A range error is thrown if the level is smaller 1 now (I assume that levels start at 1). See https://github.com/Kugelschieber/breach/pull/22/commits/00a7e8b9dcb454c846e9a2e171a68e803cd85f2a
Please resolve, if you're content with this this.